### PR TITLE
Pipeline

### PR DESCRIPTION
**Pull request by KaneFreeman**
Adding in pipeline from POC. Converted to TS, enhanced the region parser to handle none JS/TS files and added tests for the whole pipeline.

### CODE REVIEW

**Comment by agubler** [comment]
@KaneFreeman This looks great, just a few observations. It would be much easier to review if these changes could be broken up into smaller chunks. Something like:

1) Basic Pipeline
2) Dojo Components
3) Region Enhancements
4) Custom Syntax Highlighting

Then each change could have an accompanying description and goals for the specific changes.

File: scripts/rehype-prism.ts [hunk 1]
```diff
@@ -0,0 +1,55 @@
+const visit = require('unist-util-visit');
```
Comment (agubler) [inline scripts/rehype-prism.ts]: How come we need a custom syntax highlighting script?

File: scripts/interface.d.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
+import { WNode } from '@dojo/framework/widget-core/interfaces';
```
Comment (agubler) [inline scripts/interface.d.ts]: How come we need a shared interfaces file for the pipeline scripts?

File: scripts/tsconfig.json [hunk 3]
```diff
@@ -0,0 +1,30 @@
+{
```
Comment (agubler) [inline scripts/tsconfig.json]: funky formatting going on here

File: src/widgets/Aside.tsx [hunk 4]
```diff
@@ -0,0 +1,28 @@
+import WidgetBase from '@dojo/framework/widget-core/WidgetBase';
+import { tsx } from '@dojo/framework/widget-core/tsx';
+
+const AsideStyles = {
```
Comment (agubler) [inline src/widgets/Aside.tsx]: We should move all of the inline styles to css

File: content/tutorials/sample-tutorial.md [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+## Metadata
+[Metadata overview="Learn how to create and style custom widgets in Dojo.", topic="widgets"]
+
+## Aside
+[Aside title="Mandatory object for properties"]
+The 2nd argument of the `w()` function is mandatory even you have no properties to pass in. This is to ensure the correct type guarding for all widgets in TypeScript.
+[/Aside]
+
+## Task
+<dojo-task>Create a new root node for the application</dojo-task>
```
Comment (agubler) [inline content/tutorials/sample-tutorial.md]: Shouldn't this be using the `[Task....]....[/Task]` pragma over a custom element?

Comment (KaneFreeman) [reply content/tutorials/sample-tutorial.md]: Missed this one. Needed to change that back.

File: scripts/interface.d.ts (hunk 2 above)
Comment (KaneFreeman) [reply scripts/interface.d.ts]: The rehype-prism library used previously did not have an option for loading the extra languages (like jsx / tsx) we needed. So I copied their file and added in the language loading piece. I will probably submit a PR back to the actual library but this was faster and I needed to test it anyways.

File: scripts/rehype-prism.ts (hunk 1 above)
Comment (KaneFreeman) [reply scripts/rehype-prism.ts]: The rehype-prism library used previously did not have an option for loading the extra languages (like jsx / tsx) we needed. So I copied their file and added in the language loading piece. I will probably submit a PR back to the actual library but this was faster and I needed to test it anyways.

File: scripts/interface.d.ts (hunk 2 above)
Comment (KaneFreeman) [reply scripts/interface.d.ts]: Seemed easier from an organizational perspective to have them all in one place, but they can be split up if we want.

**Comment by KaneFreeman** [comment]
@agubler , I will go ahead and break this into separate PRs and make adjustments per your notes above.

**Pull request closed without merging**